airflow/contrib/operators/gcp_spanner_operator.py

```diff
@@ -197,3 +197,192 @@ def execute(self, context):
+
+    :param project_id: The ID of the project which owns the instances, tables and data.
+    :type project_id: str
+    :param instance_id: Cloud Spanner instance ID.
+    :type instance_id: str
+    :param database_id: Cloud Spanner database ID.
+    :type database_id: str
+    :param ddl_statements: String list containing DDL for the new database.
+    :type ddl_statements: [str]
+    :param gcp_conn_id: The connection ID used to connect to Google Cloud Platform.
+    :type gcp_conn_id: str
+    """
+    # [START gcp_spanner_database_deploy_template_fields]
+    template_fields = ('project_id', 'instance_id', 'database_id', 'ddl_statements',
+                       'gcp_conn_id')
+    template_ext = ('.sql', )
+    # [END gcp_spanner_database_deploy_template_fields]
+
+    @apply_defaults
+    def __init__(self,
+                 project_id,
+                 instance_id,
+                 database_id,
+                 ddl_statements,
+                 gcp_conn_id='google_cloud_default',
+                 *args, **kwargs):
+        # type: (str, str, str, [str], str, object, object) -> None
+        self.instance_id = instance_id
+        self.project_id = project_id
+        self.database_id = database_id
+        self.ddl_statements = ddl_statements
+        self.gcp_conn_id = gcp_conn_id
+        self._validate_inputs()
+        self._hook = CloudSpannerHook(gcp_conn_id=gcp_conn_id)
+        super(CloudSpannerInstanceDatabaseDeployOperator, self).__init__(*args, **kwargs)
+
+    def _validate_inputs(self):
+        if not self.project_id:
+            raise AirflowException("The required parameter 'project_id' is empty")
+        if not self.instance_id:
+            raise AirflowException("The required parameter 'instance_id' is empty")
+        if not self.database_id:
+            raise AirflowException("The required parameter 'database_id' is empty")
+        if not self.ddl_statements:
+            raise AirflowException("The required parameter 'ddl_statements' is empty")
+
+    def execute(self, context):
+        if not self._hook.get_database(self.project_id,
+                                       self.instance_id,
+                                       self.database_id):
+            self.log.info("Creating Cloud Spanner database "
+                          "'{}' in project '{}' and instance '{}'".
+                          format(self.database_id, self.project_id, self.instance_id))
+            return self._hook.create_database(project_id=self.project_id,
+                                              instance_id=self.instance_id,
+                                              database_id=self.database_id,
+                                              ddl_statements=self.ddl_statements)
+        else:
+            self.log.info("The database '{}' in project '{}' and instance '{}'"
+                          " already exists. Not modifying it.".
```

Comment:
Maybe replace `Not modifying it.` with something like `Nothing to do, aborting.`?
Because "Not modifying it" sounds like we were about to modify something, while it's always only about creating or not.

Reply:
I changed it to "Nothing to do. Exiting" . sounds better than aborting. Aborting suggests an error.